drivers/ethernet/eth_stm32_hal.c

```diff
@@ -846,6 +849,15 @@ static int eth_stm32_hal_set_config(const struct device *dev,
 				     sizeof(dev_data->mac_addr),
 				     NET_LINK_ETHERNET);
 		return 0;
+#if defined(CONFIG_NET_PROMISCUOUS_MODE)
+	case ETHERNET_CONFIG_TYPE_PROMISC_MODE:
+		dev_data = DEV_DATA(dev);
+		heth = &dev_data->heth;
```

Comment:
This could be factorized at the beginning of function.

Reply:
Agreed, I'll pull the assignments for `dev_data` and `heth` out of the switch statements for both cases and put them just under the declarations.